app/utils/request-info.ts

```diff
@@ -1,11 +1,10 @@
-import { type SerializeFrom } from '@remix-run/node'
 import { useRouteLoaderData } from '@remix-run/react'
 import { type loader as rootLoader } from '~/root.tsx'
 
 /**
  * @returns the request info from the root loader
  */
 export function useRequestInfo() {
-	const data = useRouteLoaderData('root') as SerializeFrom<typeof rootLoader>
+	const data = useRouteLoaderData<typeof rootLoader>('root')!
```

Comment:
I strongly prefer avoiding `!`. Could we use our `invariant` util instead?

Reply:
Aha, I've been working on a project that currently isn't using `invariant` so didn't jump to it.
I've added now 😊